c++/src/kj/async.c++

```diff
@@ -131,7 +131,7 @@ public:
 // =======================================================================================
 
 Canceler::~Canceler() noexcept(false) {
-  cancel("operation canceled");
+  cancel(KJ_EXCEPTION(DISCONNECTED, "operation canceled"));
 }
 
 void Canceler::cancel(StringPtr cancelReason) {
```

Comment:
Hmm, I wonder if the exception type here should change too?

Reply:
Actually... Yeah, probably we should not change the destructor, but should instead change the implementation of `cancel(StringPtr)`...

Reply:
What's the thinking behind them being different? Is it that developers should explicitly `cancel(StringPtr)` and canceling via the destructor is considered a bug or?

And we'd be back to picking a string to use in `BlockedPumpTo`'s `cancel` (which is fine).